app/Rules/ReCaptchaValidation.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+namespace App\Rules;
+
+use Illuminate\Contracts\Validation\ImplicitRule;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Str;
+
+class ReCaptchaValidation implements ImplicitRule
+{
+    /**
+     * @var string $secret Secret Site Key
+     */
+    protected $secret;
+
+    /**
+     * @var string $minScore Lowest score to pass (0.0 - 1.0)
+     */
+    protected $minScore;
+
+    /**
+     * @var string $appUrl App URL of client request
+     */
+    protected $appUrl;
+
+    public function __construct($secret, $minScore, $appUrl) {
+        $this->secret = $secret;
+        $this->appUrl = $appUrl;
+        $this->minScore = $minScore;
+    }
+
+    /**
+     * Comparison against expected hostname.
+     * 
+     * @param  string $hostname
+     * @return bool  
+     */
+    public function verifyHostname($hostname)
+    {
+        $parsedUrl = parse_url($this->appUrl);
+        $expectedHostname = Arr::get($parsedUrl, 'host', null);
+
+        if (filled($expectedHostname)) {
+            if (false === Str::of($expectedHostname)->exactly($hostname)) {
+                return false;
+            }
+        } else {
+            logger()->error(self::class.': hostname detection failed; ReCaptcha hostname verification disabled');
+        }
+
+        return true;
+    }
+
+    /**
+     * Verifies the ReCaptcha Request with the official ReCaptcha service library
+     * 
+     * @param  string $secretKey
+     * @return \ReCaptcha\Response 
+     */
+    public function verify($token)
+    {
+        $recaptchaResponse = new \ReCaptcha\Response(false);
+
+        try {
+            $recaptcha = new \ReCaptcha\ReCaptcha(
```

Comment:
I wonder if this dependency would be better injected into the constructor? This would better enable us to test it?

I think that's probably why writing the tests for this class feels tricky and perhaps why it felt necessary to expose `verify` as a public method for tinkering in the tests.

Reply:
Ah yeah that makes sense. All this code went through several iterations and at one point this class was instantiated manually in the Controllers and I wanted to minimize that setup. But now it can happen in the ServiceProvider :+1: will change it